src/s3b_the_roach_market.tw2

```diff
@@ -0,0 +1,280 @@
+<!-- If using debug menu shortcuts, prompts player to enter their name before beginning initial Roach Market sequence -->
+::RM-first-entrance-nameprompt
+{ (set: $playerName to (prompt: "Enter name:", ""))
+(if: (text: $playerName)'s length > 1)[Right, my name is $playerName. [[Moving on|RM-first-entrance]].<!-- what this code does is check the ending of the custom $playerName string to hopefully pluralize it properly -->{
+  (if: $playerName's last is "s")[
+    (set: $playerNames to ($playerName + "es"))
+  ]
+  (else-if: ($playerName's last is "h") and ($playerName's 2ndlast is "s"))[
+    (set: $playerNames to ($playerName + "es"))
+  ]
+  (else-if: ($playerName's last is "h") and ($playerName's 2ndlast is "c"))[
+    (set: $playerNames to ($playerName + "es"))
+  ]
+  (else:)[
+    (set: $playerNames to ($playerName + "s"))
+  ]
+}]
+}(else:)[No, [[that's not right|RM-first-entrance-nameprompt]].]
+
+<!-- Scene for the first time a player visits the Roach Market -->
+::RM-first-entrance
+Going into the Roach Market, the first thing everybody walks by is “The Stash”: Dopeman Dan’s head shop and drug emporium.
```

Comment:
Seems like we should be able to swing by The Stash again once we reach any of the branching "where to next?" sequences. We might need to draft something for that, even if it's just the player dismissing the idea by going "Nah, I already lost five minutes of my life that I'll never get back to him." What do you all think?

Reply:
The passage `::RM-later-entrance` on line 29 is a start for that; there isn't any logic or path to bring the player there yet though. I figure the text of that section will work as is for later return visits to the Roach Market overall, and have a similar variation of the text to display if the player swings by the Stash without yet leaving the Roach Market.

Reply:
Yeah, that seems like a fun option to me! My attempt (take any part or leave it): "There was no noise from Dan's tent. He was likely either passed out or had left briefly on one of his "snack quests" to the nearby landfill. I wasn't about to push my luck by investigating."